extension/src/popup/components/sendPayment/SendConfirm/SubmitResult/index.tsx

```diff
@@ -67,13 +84,95 @@ export const SubmitSuccess = ({ viewDetails }: { viewDetails: () => void }) => {
   } = useSelector(transactionSubmissionSelector);
   const { t } = useTranslation();
   const isSwap = useIsSwap();
+  const dispatch: AppDispatch = useDispatch();
 
   const sourceAsset = getAssetFromCanonical(asset);
+  const { recommendedFee } = useNetworkFees();
+  const publicKey = useSelector(publicKeySelector);
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+
+  const server = stellarSdkServer(networkDetails.networkUrl);
+  const isHardwareWallet = !!useSelector(hardwareWalletTypeSelector);
+
+  const removeTrustline = async (assetCode: string, assetIssuer: string) => {
+    const changeParams = { limit: "0" };
+    const sourceAccount: Account = await server.loadAccount(publicKey);
+
+    const transactionXDR = new StellarSdk.TransactionBuilder(sourceAccount, {
+      fee: xlmToStroop(recommendedFee).toFixed(),
+      networkPassphrase: networkDetails.networkPassphrase,
+    })
+      .addOperation(
+        StellarSdk.Operation.changeTrust({
+          asset: new StellarSdk.Asset(assetCode, assetIssuer),
+          ...changeParams,
+        }),
+      )
+      .setTimeout(180)
+      .build()
+      .toXDR();
+
+    const trackRemoveTrustline = () => {
+      emitMetric(METRIC_NAMES.manageAssetRemoveAsset, {
+        assetCode,
+        assetIssuer,
+      });
+    };
+
+    if (isHardwareWallet) {
+      await dispatch(startHwSign({ transactionXDR, shouldSubmit: true }));
+      trackRemoveTrustline();
+    } else {
+      await signAndSubmit(transactionXDR, trackRemoveTrustline);
+    }
+  };
+
+  const signAndSubmit = async (
+    transactionXDR: string,
+    trackChangeTrustline: () => void,
+  ) => {
+    const res = await dispatch(
+      signFreighterTransaction({
+        transactionXDR,
+        network: networkDetails.networkPassphrase,
+      }),
+    );
+
+    if (signFreighterTransaction.fulfilled.match(res)) {
+      const submitResp = await dispatch(
+        submitFreighterTransaction({
+          publicKey,
+          signedXDR: res.payload.signedTransaction,
+          networkDetails,
+        }),
+      );
+
+      if (submitFreighterTransaction.fulfilled.match(submitResp)) {
+        dispatch(
+          getAccountBalances({
+            publicKey,
+            networkDetails,
+          }),
+        );
+        trackChangeTrustline();
+        dispatch(resetSubmission());
+        navigateTo(ROUTES.account);
+      }
```

Comment:
we shouldn't need to getAccountBalances and resetSubmission, since when we load /account page it's done there. I noticed we're doing that in the ManageAssetRows/index file though, I don't think it's needed there as well

```suggestion
      if (submitFreighterTransaction.fulfilled.match(submitResp)) {
        trackChangeTrustline();
        navigateTo(ROUTES.account);
      }
```

Reply:
makes sense, thanks!